### PR TITLE
Fix README and recommended_rules about no-this-before-super

### PR DESCRIPTION
**Pull request by magurotuna**
Related PR: #147 

I forgot to add `no-this-before-super` to README and recommended_rules. Now I've done it.
In addition, I made another test to check it works in case there are more than one class.

### CODE REVIEW

File: src/rules/no_this_before_super.rs [hunk 1]
```diff
@@ -182,5 +182,23 @@ class A extends B {
...
       10,
     );
+
+    assert_lint_err_on_line::<NoThisBeforeSuper>(
+      r#"
+class A extends B {
+  constructor() {
+    super();
+  }
+}
```
[the remark this answers is not in the document]
Comment (magurotuna) [reply src/rules/no_this_before_super.rs:192]: Thanks for your comment. I missed that point.
It seems that some more implementation is required to detect incorrect codes in nested class. I'll fix it.

[the remark this answers is not in the document]
Comment (bartlomieju) [reply src/rules/no_this_before_super.rs:192]: @magurotuna yes, most likely you'll need to call `swc_ecma_visit::visit_class(self, class, parent)`. It's good to do a search - there are already rules that use this solution

[the remark this answers is not in the document]
Comment (magurotuna) [reply src/rules/no_this_before_super.rs:205]: @bartlomieju Thanks for the helpful advice again! I updated the implementation.

**Comment by magurotuna** [comment]
@bartlomieju 
I've fixed the implementation to be able to detect lint errors in nested classes and an inline super class. Would you check them please?

**Pull request merged**